rust/foxglove/src/websocket/server.rs

```diff
@@ -155,6 +157,10 @@ pub(crate) struct Server {
     tasks: parking_lot::Mutex<Option<JoinSet<()>>>,
     /// Configuration to support TLS streams when enabled.
     stream_config: StreamConfiguration,
+    /// Callback for when a client connects
```

Comment:
Ditto, use the `listener` to handle these. The struct who implements the `Listener` trait should do that